smarts/core/scenario.py

```diff
@@ -253,9 +259,14 @@ def variations_for_all_scenario_roots(scenario_roots, agents_to_be_briefed):
             agent_missions = agent_missions or [None]
             social_agents = social_agents or [None]
 
-            roll_routes = random.randint(0, len(routes))
-            roll_agent_missions = random.randint(0, len(agent_missions))
-            roll_social_agents = random.randint(0, len(social_agents))
+            if shuffle_scenarios:
+                roll_routes = random.randint(0, len(routes))
+                roll_agent_missions = random.randint(0, len(agent_missions))
+                roll_social_agents = random.randint(0, len(social_agents))
+            else:
+                roll_routes = 0
+                roll_agent_missions = 0
+                roll_social_agents = 0
```

Comment:
nit: I know it means more assignments but the tendency is to initialise a variable on the same block level.

```suggestion
            roll_routes = 0
            roll_agent_missions = 0
            roll_social_agents = 0
 
            if shuffle_scenarios:
                roll_routes = random.randint(0, len(routes))
                roll_agent_missions = random.randint(0, len(agent_missions))
                roll_social_agents = random.randint(0, len(social_agents))
```